README.md

```diff
@@ -1,141 +1,481 @@
 
-2. libclickhouseodbc.so will be at ```build/driver/libclickhouseodbc.so```
+### Building from sources: Red Hat/CentOS
+
+#### Build-time dependencies: UnixODBC <!-- omit in toc -->
 
-```bash
-mkdir -p build; cd build && cmake .. && make -j $(nproc || sysctl -n hw.ncpu || echo 4)
+Execute the following in the terminal:
+
+```sh
+sudo yum groupinstall "Development Tools"
+sudo yum install centos-release-scl
+sudo yum install devtoolset-8
+sudo yum install git cmake openssl-devel unixODBC-devel
 ```
 
-## Building (Linux Debian based .deb package)
-```bash
-sudo apt install -y devscripts debhelper cmake ninja-build lsb-release unixodbc-dev
-debuild -us -uc -i --source-option=--format="3.0 (native)"
+#### Build-time dependencies: iODBC <!-- omit in toc -->
+
+Execute the following in the terminal:
+
+```sh
+sudo yum groupinstall "Development Tools"
+sudo yum install centos-release-scl
+sudo yum install devtoolset-8
+sudo yum install git cmake openssl-devel libiodbc-devel
 ```
-or
-```bash
-sudo apt install -y sudo pbuilder fakeroot debhelper debian-archive-keyring debian-keyring
-sudo pbuilder create --configfile debian/.pbuilderrc && pdebuild --configfile debian/.pbuilderrc
+
+#### Build steps <!-- omit in toc -->
+
+All of the following commands have to be issued right after this one command issued in the same terminal session:
+
+```sh
+scl enable devtoolset-8 -- bash
 ```
 
+Clone the repo with submodules:
 
-## Building (windows visual studio)
-```bat
-cd vs && build_vs.bat
+```sh
+git clone --recursive git@github.com:ClickHouse/clickhouse-odbc.git
 ```
 
-## Building (windows cmake) (Developer only: setup window still not working)
-```bat
-cd vs && build_cmake.bat
+Enter the cloned source tree, create a temporary build folder, and generate a Makefile for the project in it:
+
+```sh
+cd clickhouse-odbc
+mkdir build
+cd build
+
+# Configuration options for the project can be specified in the next command in a form of '-Dopt=val'
+
+cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo ..
+```
+
+Build the generated solution in-place:
+
+```sh
+cmake --build . -C RelWithDebInfo
+cmake --build . -C RelWithDebInfo --target package
+```
+
+...and, optionally, run tests (note, that for non-unit tests, preconfigured driver and DSN entries must exist, that point to the binaries generated in this build folder):
+
+```sh
+cmake --build . -C RelWithDebInfo --target test
 ```
 
-## Build with tests (needs configured ~/.odbc.ini with DSN=clickhouse_localhost)
-```bash
-mkdir -p build; cd build
-( cd ../contrib && git clone https://github.com/nanodbc/nanodbc )
-cmake -G Ninja -DTEST_DSN=clickhouse_localhost -DCMAKE_BUILD_TYPE=Debug -DUSE_DEBUG_17=1 .. && ninja
-ctest -V
+### Building from sources: Debian/Ubuntu
+
+#### Build-time dependencies: UnixODBC <!-- omit in toc -->
+
+Execute the following in the terminal:
+
+```sh
+sudo apt install build-essential git cmake libpoco-dev libssl-dev unixodbc-dev
```

Comment:
libpoco?

Reply:
yes, for cases when the driver is built with unboundled poco.

Reply:
In general, I want to list those deps that will cover building and running all possible types of builds.